ensemble/random_forest.h

```diff
@@ -0,0 +1,55 @@
+#ifndef _MACHINELEARNING_ENSEMBLE_RANDOM_FOREST_H_
+#define _MACHINELEARNING_ENSEMBLE_RANDOM_FOREST_H_
+
+#include "tree/tree.h"
+// #include "base/learner.h"
+
+namespace ensemble {
+/**
+ * Base class for Random Forest Classification and Regression algorithms.
+ */
+class RandomForest /*: public base::Learner */{
+  
+  public:
+    
+    /**
+     * Standard Constructor.
+     * \param tree_count Number of trees.
+     * \param thread_count Number of threads.
+     */
+    RandomForest(int tree_count, int thread_count);
+
+    /**
+     * \brief Standard destructor.
+     */
+    ~RandomForest();
+
+    /**
+     * Train the learner with a given dataset.
+     * \param X A pointer object of Collection.
+     */  
+    void Train(/*const Collection *X*/);
+
+    /**
```

Comment:
Lets handle OOB in a later pull request. Remove these comments and we will be good to go.